viewer/src/scripts/components/LeafletMap.jsx

```diff
@@ -40,47 +50,102 @@ var LeafletMap = React.createClass({
   map: null, 
   layer: null,
 
+  getInitialState: function () {
+    return {'value' : 0.5}
+  },
+
   componentDidMount: function () {    
     this.map = L.map(this.getDOMNode());
 
     baseLayers['Azavea'].addTo(this.map);    
     this.map.lc = L.control.layers(baseLayers).addTo(this.map);
     this.map.setView([30.25, -97.75], 4);
-
     this_map = this.map;
   },
 
-  render: function() {      
+  shouldComponentUpdate: function(nextProps, nextState) {
+    if(nextState.opacity) {  //if changing opacity
+      if(nextState.value !== this.state.value && this.layer) {
+        this.layer.setOpacity(nextState.value);
+      }
+      nextState.opacity = false;  //only allow handleOpacityUpdate to change opacity through Slider
+      return false;
+    } 
+    return true;
+  },
+
+  handleOpacityUpdate: function(event) {
+    this.setState({
+      value: event,
+      opacity: true
+    });
+  },
+
+  handleUpdateLayer: function() {
     var active = this.props.active; 
     var entry = this.props.active.entry;
-
-    if (this.isMounted() && active.entry) {      
-      this.map.setView([entry.center[1], entry.center[0]], entry.layer.zoom);
-      
+    this.map.setView([entry.center[1], entry.center[0]], entry.layer.zoom);
       var removeLayer = function (map, oldLayer){
         if (oldLayer) { 
           map.removeLayer(oldLayer) 
           map.lc.removeLayer(oldLayer);
         }
-
-
       }
-      
-      var oldLayer = this.layer;    
-      var args = active.band;      
-      args['breaks'] = active.entry.breaks.join(',');      
+     
+      var args = active.band; 
+      args['breaks'] = active.entry.breaks.join(',');
+
+      if(this.layer) {
+        var tmsUrl = this.layer._url
+        var findColorRegex = /&colorRamp=/;
+        tmsUrlSplit = tmsUrl.split(findColorRegex);
+        if(tmsUrlSplit[1]) {
+          args['colorRamp'] = tmsUrlSplit[1];
+        }   
+      }
+ 
+      var oldLayer = this.layer; 
       var url = this.props.tmsUrl + "/" + active.entry.layer.name  + "/{z}/{x}/{y}?" + $.param( args );
-      var newLayer = L.tileLayer(url, {minZoom: 1, maxZoom: 12, tileSize: 256, tms: false, opacity: 0.95});
+      var newLayer = L.tileLayer(url, {minZoom: 1, maxZoom: 12, tileSize: 256, tms: false, opacity: this.state.value});
       newLayer.addTo(this.map);
       this.map.lc.addOverlay(newLayer, entry.layer.name);
       this.layer = newLayer;
-
       _.delay(removeLayer, 1000, this.map, oldLayer);
+  },
 
+  handleUpdateColorRamp: function() {
+    var active = this.props.active; 
+    var tmsUrl = this.layer._url
+    var findColorRegex = /&colorRamp/;
+    var tmsUrlSplit = tmsUrl.split(findColorRegex);
```

Comment:
another kludge here which potentially breaks other query parameters if they come after `colorRamp` on the URL.  this only works as desired if `colorRamp` is the last query parameter on the URL.